frontends/systolic-lang/gen_post_op.py

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/env python3
+
+import calyx.builder as cb
+from calyx import py_ast
+from gen_array_component import NAME_SCHEME
+from gen_pe import BITWIDTH, INTWIDTH, FRACWIDTH
+from fud.stages.verilator import numeric_types
+from calyx.utils import float_to_fixed_point
+
+
+# Name of the ouput array
+OUT_MEM = "out_mem"
+DEFAULT_POST_OP = "default_post_op"
+LEAKY_RELU_POST_OP = "leaky_relu_post_op"
+COND_REG = "cond_reg"
+WRITE_DONE_COND = "write_done_cond"
+
+
+def add_register_params(comp: cb.ComponentBuilder, name, width):
+    """
+    Add params to component `comp` if we want to use a register named
+    `name` inside `comp.`Specifically adds the write_en, in, and out ports.
+    """
+    comp.output(f"{name}_write_en", 1)
+    comp.output(f"{name}_in", width)
+    comp.input(f"{name}_out", width)
+
+
+def add_write_mem_params(comp: cb.ComponentBuilder, name, addr_width):
+    """
+    Add arguments to component `comp` if we want to write to a mem named `name` with
+    width of `addr_width` inside `comp.`
+    """
+    comp.output(f"{name}_addr0", addr_width)
+    comp.output(f"{name}_write_data", BITWIDTH)
+    comp.output(f"{name}_write_en", 1)
+    comp.input(f"{name}_done", 1)
+
+
+def add_systolic_input_params(comp: cb.ComponentBuilder, row_num, addr_width):
+    """
+    Add ports "r_{row_num}_valid", "r_{row_num}_value", "r_{row_num}_idx" to comp.
+    These ports are meant to read from the systolic array output.
+    """
+    comp.input(NAME_SCHEME["systolic valid signal"].format(row_num=row_num), 1)
```

Comment:
This name scheme thing has always been a terrible back and probably unmaintainable in the long-term. It was needed before we had the `builder` stuff but we can probably get rid of this now. Any thoughts on how hard it would be?